pkg/observ/observ.go

```diff
@@ -61,11 +62,20 @@ func Tracer(service string) buffalo.MiddlewareFunc {
 				trace.WithSpanKind(trace.SpanKindClient))
 			defer span.End()
 
+			handler := next(&observabilityContext{Context: ctx, spanCtx: spanCtx})
+
+			// Add request attributes
 			span.AddAttributes(
 				requestAttrs(ctx.Request())...,
 			)
 
-			return next(&observabilityContext{Context: ctx, spanCtx: spanCtx})
+			// SetSpan Status from response
+			if resp, ok := ctx.Response().(*buffalo.Response); ok {
+				span.SetStatus(ochttp.TraceStatus(resp.Status, ""))
+				span.AddAttributes(trace.Int64Attribute("http.status_code", int64(resp.Status)))
```

Comment:
why do you we need this line when we set the status above?

Reply:
Precisely for the reason you mentioned! TraceStatus do not have a 1-1 relationship with http.status_code, so we should save http.status_code as an attribute so that we can look and introspect what the status_code was exactly

Reply:
@manugupt1 we should have a clear distinction between tracing, metrics, and logging. The HTTP status code should be in the logging part. We shouldn't have metrics be a place for debugging biz logic, it should be a place for debugging latency. I know the stuff is very "grey" but we should try our best to have clear paths for debugging different kinds of problems :) let me know what you think!

Reply:
So, I did a bit of research. All existing tracing libraries both opencensus and opentracing seem to suggest storing http status code.

http://opentracing.io/documentation/pages/instrumentation/common-use-cases.html
https://opencensus.io/core-concepts/tracing/

My inclination is to put it in there.

Another thing is that traces should not have logs and everything, but traces should be able to point out to the correct logs. So, when you look at the trace, you are able to reach the relevant logs asap rather than having to debug trace individually and logs individually. This is about how we approach problems. But I think that will be a nice way to look at the problems anyway even when the traces are sampled.

Reply:
@manugupt1 -- Ah. I didn't mean traces should have logs. Logs can have a trace ID and so when you look at a trace and want to know more information about that request (such as http status, module name/version, etc), you can go to the logs and look up a log by that same trace ID.

Otherwise, here's my main question: 

- What determines *which* request data go into the trace metadata and *which* request data go into the log fields? Does http status live in both places? Does an error message live only in logs but not in trace metadata? Should operations go in tracing but not in logs? Should error kind live in both places or just logs? etc etc etc 

For now I think it's totally fine to have http status in both places but I hope you see what I mean in that things can start to get unclear about *where* data should flow.

Will merge now!

Reply:
Yeah.. for the second one I do not have any answers as of now! I am hoping someone (probably you) who will end up using out observability stack will say this is unnecessary and we can start removing it. Thanks for merging this.